dig.go

```diff
@@ -498,9 +538,25 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) error {
 		}
 		c.isVerifiedAcyclic = true
 	}
-
 	c.nodes = append(c.nodes, n)
 
+	// Record introspection info for caller if Info option is specified
+	if info := opts.Info; info != nil {
+		params := n.ParamList().DotParam()
+		results := n.ResultList().DotResult()
+
+		info.ID = (ID)(n.id)
+		info.Inputs = make([]*Input, len(params))
+		info.Outputs = make([]*Output, len(results))
+
+		for i, param := range params {
+			info.Inputs[i] = &Input{t: param.Type}
```

Comment:
Maybe, 

```
info.Inputs[i] = &Input{
    t: param.Type,
    optional: param.Optional,
    name: param.Name,
    group: param.Group,
}
```

And the `String()` for Input then needs to account for some of this information. We can pick whatever format we prefer for it. I'm leaning something like:

```
$type[optional, name = "$name", group = "$group"]
```

Where the `optional`, `name = "$name"` or `group = "$group"` is added only if those values are set, and the `[`, `]` is added only if at least one entry was added. So  the following are all valid:

```
*db.Conn
*zap.Logger[optional]
*db.Conn[optional, name = "readonly"]
[]http.Handler[group = "http"]
```